openpgp/clearsign/clearsign.go

```diff
@@ -214,15 +214,30 @@ type dashEscaper struct {
 	config      *packet.Config
 }
 
+// Auxiliary struct to optimize error checking (from
+// https://blog.golang.org/errors-are-values)
+type errWriter struct {
+	w   io.Writer
+	err error
+}
+
+func (ew *errWriter) write(buf []byte) {
+	if ew.err != nil {
+		return
+	}
+	_, ew.err = ew.w.Write(buf)
+}
+
 func (d *dashEscaper) Write(data []byte) (n int, err error) {
+	ew := &errWriter{w: d.toHash}
```

Comment:
I would perhaps call this `toHash`, so that in the code below, it's still clear that it's hashing something?

Reply:
Yes, that's better, thanks